modules/stack/main.tf

```diff
@@ -215,3 +217,12 @@ resource "spacelift_context_attachment" "attachment" {
   stack_id   = spacelift_stack.default[0].id
   priority   = each.value
 }
+
+resource "spacelift_space" "default" {
+  count = var.dedicated_space ? 1 : 0
+
+  name            = var.component_name
+  parent_space_id = var.parent_space_id
```

Comment:
Can we add description and labels too ?

https://registry.terraform.io/providers/spacelift-io/spacelift/0.1.31/docs/resources/space#description

Reply:
All done and tested @nitrocode